### ISSUE TITLE
Modify a process instance by terminating all element instances of a flow scope

### ISSUE THREAD

**Issue by saig0** [camunda/zeebe#9691]
## Description

We want to go one step further and allow the termination of all element instances of a flow scope. As a result, the instance of the flow scope should be terminated as well. The flow scope can't be active if it doesn't have any active element instances anymore. 

The termination should be propagated to the flow scope of the terminated element instance recursively to ensure that all flow scopes are terminated if they don't contain any active element instances. The propagation ends on the process instance scope. If the process instance doesn't have any active element instances then the process instance is terminated as well.   

For example:
![image](https://user-images.githubusercontent.com/4305769/177326871-c5d29917-e8e4-4d99-83cc-c3f98bfaadc0.png)

* check the flow scopes after all terminate instructions are applied
* check each flow scope of the terminated element instances
  * terminate the instance of the flow scope (according to #9690) if it has no active element instances (and no pending sequence flows to take)
  * do the same check for the next upper flow scope if the flow scope was terminated
* verify that 
  * the process instance terminates the flow scope if all element instances are terminated
  * the process instance doesn't complete the flow scope if the element instances are terminated (i.e. doesn't take the outgoing sequence flow)
  * the process instance doesn't complete the flow scope if it has pending sequence flows to take (i.e. outgoing sequence flow is taken and activate command is written but not processed)
  * the process instance is terminated if all element instances are terminated

Out of scope:
* termination of all element instances of a process instance that is created by a call activity

Blocked by 
- #9690
